rom/dev/tests/test_image_validation.rs

```diff
@@ -731,6 +887,150 @@ fn test_runtime_digest_mismatch() {
+#[test]
+fn test_runtime_invalid_entry_point_before_iccm() {
+    let (mut hw, mut image_bundle) = helpers::build_hw_model_and_image_bundle(Fuses::default());
+    let mut output = vec![];
+
+    let image = update_entry_point(&mut image_bundle, false, ICCM_START_ADDR - 4);
+    assert_eq!(
+        ModelError::MailboxCmdFailed,
+        hw.upload_firmware(&image).unwrap_err()
+    );
+    let result = hw.copy_output_until_non_fatal_error(RUNTIME_ENTRY_POINT_INVALID, &mut output);
+    assert!(result.is_ok());
+}
+
+#[test]
+fn test_runtime_invalid_entry_point_after_iccm() {
+    let (mut hw, mut image_bundle) = helpers::build_hw_model_and_image_bundle(Fuses::default());
+    let mut output = vec![];
+
+    let image = update_entry_point(&mut image_bundle, false, ICCM_END_ADDR + 1);
+    assert_eq!(
+        ModelError::MailboxCmdFailed,
+        hw.upload_firmware(&image).unwrap_err()
+    );
+    let result = hw.copy_output_until_non_fatal_error(RUNTIME_ENTRY_POINT_INVALID, &mut output);
+    assert!(result.is_ok());
+}
+
+#[test]
+fn test_runtime_entry_point_unaligned() {
+    let (mut hw, mut image_bundle) = helpers::build_hw_model_and_image_bundle(Fuses::default());
+    let mut output = vec![];
+    let entry_point = image_bundle.manifest.runtime.entry_point;
+    let image = update_entry_point(&mut image_bundle, false, entry_point + 1);
+    assert_eq!(
+        ModelError::MailboxCmdFailed,
+        hw.upload_firmware(&image).unwrap_err()
+    );
+    let result = hw.copy_output_until_non_fatal_error(RUNTIME_ENTRY_POINT_UNALIGNED, &mut output);
+    assert!(result.is_ok());
+}
+
+#[test]
+fn test_runtime_svn_greater_than_64() {
+    let fuses = caliptra_hw_model::Fuses {
+        anti_rollback_disable: false,
+        ..Default::default()
+    };
+    let (mut hw, mut image_bundle) = helpers::build_hw_model_and_image_bundle(fuses);
+    let mut output = vec![];
+
+    let image = update_svn(&mut image_bundle, false, 65);
+    assert_eq!(
+        ModelError::MailboxCmdFailed,
+        hw.upload_firmware(&image).unwrap_err()
+    );
+    let result =
+        hw.copy_output_until_non_fatal_error(RUNTIME_SVN_GREATER_THAN_MAX_SUPPORTED, &mut output);
+    assert!(result.is_ok());
+}
+
+#[test]
+fn test_runtime_svn_less_than_min_svn() {
+    let fuses = caliptra_hw_model::Fuses {
+        anti_rollback_disable: false,
+        ..Default::default()
+    };
+    let (mut hw, mut image_bundle) = helpers::build_hw_model_and_image_bundle(fuses);
+    let mut output = vec![];
+
+    let svn = image_bundle.manifest.runtime.svn;
+    let image = update_min_svn(&mut image_bundle, false, svn + 1);
+    assert_eq!(
+        ModelError::MailboxCmdFailed,
+        hw.upload_firmware(&image).unwrap_err()
+    );
+    let result =
+        hw.copy_output_until_non_fatal_error(RUNTIME_SVN_LESS_THAN_MIN_SUPPORTED, &mut output);
+    assert!(result.is_ok());
+}
+
+#[test]
+fn test_runtime_svn_less_than_fuse_svn() {
+    let fuse_svn: [u32; 4] = [10, 0, 0, 0];
```

Comment:
Is this a valid SVN fuse value? The binary value (0b1010) does not have consecutive ones.

Reply:
Updated.